tests/select_for_update/tests.py

```diff
@@ -152,6 +176,55 @@ def test_unsupported_skip_locked_raises_error(self):
             with transaction.atomic():
                 Person.objects.select_for_update(skip_locked=True).get()
 
+    @skipIfDBFeature('has_select_for_update_of')
+    @skipUnlessDBFeature('has_select_for_update')
+    def test_unsupported_of_raises_error(self):
+        """
+        NotSupportedError is raised if a SELECT...FOR UPDATE OF... is run on
+        a database backend that supports FOR UPDATE but not OF.
+        """
+        with self.assertRaisesMessage(NotSupportedError, 'FOR UPDATE OF is not supported on this database backend.'):
+            with transaction.atomic():
+                Person.objects.select_for_update(of=('self',)).get()
+
+    @skipUnlessDBFeature('has_select_for_update', 'has_select_for_update_of')
+    def test_unrelated_of_argument_raises_error(self):
+        """
+        DatabaseError is raised if a non-relation field is specified in
+        of=(...).
+        """
+        for name in ['nonexistent', 'name', 'born__nonexistent', 'born__name']:
+            with self.subTest(name=name):
+                with self.assertRaisesMessage(
```

Comment:
We generally use this indentation style for `assertRaisesMessage()`:
https://github.com/django/django/blob/3297dede7fce4b190f7b3bf0b0fc29a734151b61/tests/many_to_one/tests.py#L111-L116